environments/servlet/core/src/main/java/org/jboss/weld/environment/jetty/LegacyWeldDecorator.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2010, Red Hat, Inc., and individual contributors
+ * by the @authors tag. See the copyright.txt in the distribution for a
+ * full listing of individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.weld.environment.jetty;
+
+import javax.servlet.ServletContext;
+
+import org.eclipse.jetty.server.handler.ContextHandler;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.jetty.util.DecoratedObjectFactory;
+import org.eclipse.jetty.util.Decorator;
+
+/**
+ * Jetty Eclipse Weld support for jetty &gt;=9 and &lt;= 9.4.20
+ *
+ * <p>This decorator relies on the the following Jetty APIs to be exposed
+ * to the webapp:<ul>
+ *     <li>org.eclipse.jetty.server.handler.ContextHandler</li>
+ *     <li>org.eclipse.jetty.servlet.ServletContextHandler</li>
+ *     <li>org.eclipse.jetty.util.DecoratedObjectFactory</li>
+ *     <li>org.eclipse.jetty.util.Decorator</li>
+ * </ul>
+ *
+ * @author <a href="mailto:ales.justin@jboss.org">Ales Justin</a>
+ */
+public class LegacyWeldDecorator extends WeldDecorator implements Decorator {
```

Comment:
I am thinking whether it's worth to keep legacy stuff around or whether we should just say that next Weld version supports Jetty 9.4.20+. Currently our docs mention 9.4.14 but we have done similar micro/minor version bumps previously within Weld's micro releases (Weld 3.1.0 -> 3.1.1 changed Jetty support from 9.3.6 to 9.4.14). Personally, I would lean towards dropping the legacies...
WDYT @mkouba?

Reply:
Obviously we could only do that once the new Jetty supporting this approach gets released.

Reply:
Yeh, this will only work on jetty-9.4.20, which gives a pretty hard change for Weld releases on a point release.  A lot of users might want to upgrade Weld prior to upgrading Jetty, so having legacy support (perhaps deprecated) for a while would lessen the impact on users.   Note also it is kind of useful for checking the signature of the non legacy decorator !